tools/system_libs.py

```diff
@@ -1724,6 +1724,50 @@ def variations(cls):
             [dict(malloc='emmalloc-verbose', **combo) for combo in combos if combo['verbose'] and not combo['memvalidate']])
 
 
+class libmimalloc(MTLibrary):
+  name = 'libmimalloc'
+
+  cflags = [
+    '-fno-builtin',
+    '-Wno-deprecated-pragma',
+    # build emmalloc as only a system allocator, without exporting itself onto
+    # malloc/free in the global scope
+    '-DEMMALLOC_NO_STD_EXPORTS',
+    # build mimalloc with an override of malloc/free
+    '-DMI_MALLOC_OVERRIDE',
+    # TODO: add build modes that include debug checks 1,2,3
+    '-DMI_DEBUG=0',
+  ]
+
+  # malloc/free/calloc are runtime functions and can be generated during LTO
+  # Therefor they cannot themselves be part of LTO.
+  force_object_files = True
+
+  includes = ['system/lib/mimalloc/include']
+
+  # Build all of mimalloc, and also emmalloc which is used as the system
+  # allocator underneath it.
+  src_dir = 'system/lib/'
+  src_files = glob_in_path(
+    path='system/lib/mimalloc/src',
+    glob_pattern='*.c',
+    # mimalloc includes some files at the source level, so exclude them here.
+    excludes=[
+      'alloc-override.c',
+      'page-queue.c',
+      'static.c',
+    ]
+  ) + files_in_path(
+    path='system/lib/mimalloc/src/prim',
+    filenames=['prim.c']
+  ) + files_in_path(
+    path='system/lib/',
+    filenames=['emmalloc.c'])
```

Comment:
Elsewhere in this file we tend to put each `+` in its on statement.  I think that will make it less messy:

e.g.:

```
    libc_files += files_in_path(                                                                     
        path='system/lib/libc/musl/src/legacy',                                                      
        filenames=['getpagesize.c', 'err.c'])                                                        
                                                                                                     
    libc_files += files_in_path(                                                                     
        path='system/lib/libc/musl/src/linux',                                                       
        filenames=['getdents.c', 'gettid.c', 'utimes.c'])                                            
                                                                                                     
    libc_files += files_in_path(                                                                     
        path='system/lib/libc/musl/src/sched',                                                       
        filenames=['sched_yield.c'])                                                                 
                                                                                                     
    libc_files += files_in_path(                                                                     
        path='system/lib/libc/musl/src/exit',                                                        
        filenames=['_Exit.c', 'atexit.c', 'at_quick_exit.c', 'quick_exit.c'])                        
                                                                                                     
    libc_files += files_in_path(                                                                     
        path='system/lib/libc/musl/src/ldso',                                                        
        filenames=['dladdr.c', 'dlerror.c', 'dlsym.c', 'dlclose.c'])       
    ```